### ISSUE TITLE
Include bundletool in the `flutter_tools`

### ISSUE THREAD

**Issue by GaryQian** [flutter/flutter#117127]
Android Appbundles (aabs) are the required format for play store app submissions. However, .aab files built by `flutter build appbundle` cannot be easily previewed.

https://github.com/google/bundletool Is required to handle bundles. To make the user experience more seamless, we should provide a way for devs to access bundletool without having to go install it themselves, whether that be wrapping bundletool commands or just providing access to the bundletool directly.

**Comment by reidbaker** [comment]
@mariamhas  fyi

**Comment by eseidel** [comment]
We're considering building this ourselves in `shorebird`: https://github.com/shorebirdtech/shorebird/issues/302  I'd love to add it to `flutter run` itself, but I fear there might be other fallout (no clue how split apks deal with Flutter's debug-time asset resolution, etc).

**Comment by eseidel** [comment]
I doubt other devs have the same fragility that `shorebird` did in split-apks vs. not, but it's just pretty silly that the dev mode for Flutter uses an apk format (universal apks instead of split apks) which will not be seen in the wild by 99% of users.

**Comment by reidbaker** [comment]
If you are willing to solve this bug I think it is something we would accept in a PR. FWIW I do not think that using universal apks for developers is a silly decision. It is a pretty normal to build the most universally runnable apk for development then do your testing/dogfooding with a release signed flavor of your app.

**Comment by eseidel** [comment]
Sorry the "silly" if anything was more self-deprecating.  I built the initial system so many years ago, I watched the ecosystem transition to appbundles, but not until living it again as a flutter-user did I realize that us still using a not-store-like-config in development might pose problems for developers. :) I'll see what I can hack together in `shorebird` then happy to consider a patch upstream!

**Comment by sarn0ld** [comment]
@eseidel with https://github.com/shorebirdtech/shorebird/issues/302 closed, you seem to be the expert to me: would it be hard to use bundle tool in flutter run?

**Comment by eseidel** [comment]
It would not be hard to bundle/expose bundletool in `flutter`, no.  And if `flutter` did, we could stop doing so at the `shorebird` layer.  We recently added `shorebird get-apk` which downloads an aab from our storage and then runs `bundle-tool build-apks`.  I think it could be interesting to have a `flutter bundle-tool` command which just exposed `bundle-tool`, but really the silliness is that Android Studio doesn't expose bundle-tool (they expect you to pull it down via gradle afaict).

**Comment by sarn0ld** [comment]
I think it would be interesting if we could `flutter run` with apks from aab files, because dynamic Gradle modules like asset packs (https://developer.android.com/guide/playcore/asset-delivery) seem to be excluded from normal APK builds.

It seems that when Android Studio is installed, we can use this Gradle task to build an APK from app bundle:  
`./gradlew app:extractApksFor{Flavor}Debug` 

I don't know the internals of `flutter run`, but maybe it is using the Gradle task `assemble{Flavor}Debug` somewhere and that could just be replaced?

**Comment by reidbaker** [comment]
@sarn0ld you are correct we run gradle tasks underneath the hood of flutter run. That said our automated testing is not at the point where I am comfortable rewriting that code wholesale and shipping it. 

I do think we should probably prototype a flag that replaces most apk building with building an aab then using bundle tool.
